krm-functions/lib/nad/v1/nad.go

```diff
@@ -0,0 +1,366 @@
+	for _, plugin := range existingNadConfig.Plugins {
+		if plugin.Type == TuningType {
+			continue
+		} else {
+			return plugin.Master, nil
+		}
+	}
+	return "", nil
+}
+
+func (r *NadStruct) GetIpamAddress() ([]Addresses, error) {
+	existingNadConfig, err := r.GetNadConfig()
+	if err != nil {
+		return []Addresses{}, err
+	}
+	for _, plugin := range existingNadConfig.Plugins {
+		if plugin.Type == TuningType {
+			continue
+		} else {
+			return plugin.Ipam.Addresses, nil
+		}
+	}
+	return []Addresses{}, nil
+}
+
+func (r *NadStruct) GetNadConfig() (NadConfig, error) {
+	nadConfigStruct := NadConfig{}
+	configSpec := r.GetConfigSpec()
+	if configSpec == "" {
+		configSpec = "{}"
+	}
+	if err := json.Unmarshal([]byte(configSpec), &nadConfigStruct); err != nil {
+		return nadConfigStruct, fmt.Errorf("invalid NAD Config, %s", err)
+	}
+	nadConfigStruct.CniVersion = CniVersion
+	if r.CniSpecType == VlanAllocOnly {
+		return nadConfigStruct, nil
+	} else if r.CniSpecType == IpVlanType {
+		if nadConfigStruct.Plugins == nil || len(nadConfigStruct.Plugins) == 0 {
+			nadConfigStruct.Plugins = []PluginCniType{
+				{
+					Capabilities: Capabilities{
+						Ips: true,
+					},
+					Mode: ModeL2,
+					Ipam: Ipam{
+						Type: NadType,
+						Addresses: []Addresses{
+							{},
+						},
+					},
+				},
+			}
+		}
+		return nadConfigStruct, nil
+	} else if r.CniSpecType == MacVlanType {
+		if nadConfigStruct.Plugins == nil || len(nadConfigStruct.Plugins) == 0 {
+			nadConfigStruct.Plugins = []PluginCniType{
+				{
+					Capabilities: Capabilities{Ips: true},
+					Mode:         ModeBridge,
+					Ipam: Ipam{
+						Type: NadType,
+						Addresses: []Addresses{
+							{},
+						},
+					},
+				},
+				{
+					Capabilities: Capabilities{Mac: true},
+					Type:         TuningType,
+				},
+			}
+		}
+		return nadConfigStruct, nil
+	} else if r.CniSpecType == SriovType {
+		if nadConfigStruct.Plugins == nil || len(nadConfigStruct.Plugins) == 0 {
+			nadConfigStruct.Plugins = []PluginCniType{
+				{
+					Capabilities: Capabilities{Ips: true},
+					Mode:         ModeBridge,
+					Ipam: Ipam{
+						Type: NadType,
+						Addresses: []Addresses{
+							{},
+						},
+					},
+				},
+			}
+		}
+		return nadConfigStruct, nil
+	} else if r.CniSpecType == OtherType {
+		if nadConfigStruct.Plugins == nil || len(nadConfigStruct.Plugins) == 0 {
+			nadConfigStruct.Plugins = []PluginCniType{
+				{
+					Capabilities: Capabilities{
+						Ips: true,
+					},
+					Mode: ModeBridge,
+					Ipam: Ipam{
+						Type: NadType,
+						Addresses: []Addresses{
+							{},
+						},
+					},
+				},
+			}
+		}
+		return nadConfigStruct, nil
+	}
+	return nadConfigStruct, nil
+}
+
+// SetConfigSpec sets the spec attributes in the kubeobject according the go struct
+func (r *NadStruct) SetConfigSpec(spec *nadv1.NetworkAttachmentDefinitionSpec) error {
+	return r.K.SetNestedString(spec.Config, ConfigType...)
+}
+
+func (r *NadStruct) SetNadConfig(config NadConfig) error {
```

Comment:
Again, convention but should getters and setters be placed together for readability?
ie 
getA()
setA()
getB()
setB()

Reply:
Aligned the functions similar to other libs..